packages/common/src/authentication/AuthContext.tsx

```diff
@@ -124,20 +121,27 @@ export const AuthProvider: FC = ({ children }) => {
 
   // returns MRU store, if set
   // or the first store in the list
-  // TODO: return default store rather than first - currently not provided by API
-  const getStore = async () => {
-    if (mostRecentlyUsedCredentials?.store)
+  const getStore = async (token?: string) => {
+    const userDetails = await getStores(token);
+    const defaultStore = userDetails?.defaultStore;
+    const stores = userDetails?.stores.nodes;
+
+    if (
+      mostRecentlyUsedCredentials?.store &&
+      stores?.some(store => store.id === mostRecentlyUsedCredentials?.store?.id)
+    ) {
       return mostRecentlyUsedCredentials.store;
+    }
 
-    const { nodes } = await getStores();
-    return !!nodes && nodes?.length && nodes?.length > 0
-      ? nodes?.[0]
-      : undefined;
+    if (!!defaultStore) return defaultStore;
+
+    return !!stores && stores?.length > 0 ? stores?.[0] : undefined;
   };
 
   const login = async (username: string, password: string) => {
     const { token, error } = await mutateAsync({ username, password });
-    const store = await getStore();
+    setHeader('Authorization', `Bearer ${token}`);
```

Comment:
adding this as a catch for subsequent queries which need to use the new token and might be called before the cookie is updated